src/components/Todo/Todo.js

```diff
@@ -1,70 +1,170 @@
+    const newItemList = [...items, {
+        value,
+        isDone: false,
+        id: newCount,
+        disabled: true,
+        hide: hide
+        }
+      ]
+    setItems(newItemList);
+    setCount(newCount);
+    saveToLocalStorage(newItemList);
+    localStorage.setItem("count", newCount);
   };
-  
+
+  const onEditItem = id => {
+    const newItemList = items.map(item => {
+      const newItem = { ...item };
+      if (item.id === id) {
+        newItem.disabled = false;
+      }
+      return newItem;
+    });
+    setItems(newItemList);
+    saveToLocalStorage(newItemList);
+  };
+
+  const onBlurItem = (id, newValue) => {
+    const newItemList = items.map(item => {
+      const newItem = { ...item };
+      if (item.id === id) {
+        newItem.value = newValue;
+        newItem.disabled = true;
+      }
+      return newItem;
+    });
+    setItems(newItemList);
+    saveToLocalStorage(newItemList);
+  };
+
+  const onClickDeleteCompleted = () => {
+    const newItemList = items.filter(item => item.isDone == false);
+    setItems(newItemList);    
+    saveToLocalStorage(newItemList);
+  };
+
+  const hideCompleted = () => {
+    const newItemList = items.map(item => {
+      const newItem = { ...item };
+      if (item.isDone === true) {
+        newItem.hide = true;
+      } else {
+          newItem.hide = false;
+        }
+      return newItem;
+    });
+    setItems(newItemList);
+    setBottomNavigationValue ("active");
+    saveToLocalStorage(newItemList);
+  }
+
+  const hideActive = () => {
+    const newItemList = items.map(item => {
+      const newItem = { ...item };
+      if (item.isDone === false) {
+        newItem.hide = true;
+      } else newItem.hide = false;
+      return newItem;
+    });
+    setItems(newItemList);
+    setBottomNavigationValue ("completed");
+    saveToLocalStorage(newItemList);
+  }
+
+  const displayAll = () => {
+    const newItemList = items.map(item => {
+      const newItem = { ...item };
+        newItem.hide = false;
+      return newItem;
+    });
+    setItems(newItemList);  
+    setBottomNavigationValue ("all");
+    saveToLocalStorage(newItemList);
+  }
+
   return (
     <>
       <h1 className={styles.title}>Задачи на сегодня:</h1>
-      <InputItem onClickAdd={onClickAdd} />
-      <ItemList items={items} onClickDone={onClickDone} onClickDelete={onClickDelete} />
-      <Footer count={count} />
+      <InputItem onClickAdd={onClickAdd} items={items} />
+      <ItemsContext.Provider value={{onClickDone, onClickDelete, onEditItem, onBlurItem}}>
+        <DndProvider backend={HTML5Backend}>
+          <ItemList  items={items} moveItems={moveItems}/>
+        </DndProvider>
+      </ItemsContext.Provider>
+      <Footer bottomNavigationValue={bottomNavigationValue} count={items.filter(item => item.isDone == false).length} onClickDeleteCompleted={onClickDeleteCompleted} hideCompleted={hideCompleted} hideActive={hideActive} displayAll={displayAll}/>
```

Comment:
Codacy found an issue: [Expected '===' and instead saw '=='.](https://app.codacy.com/gh/WebHeroSchool/agatta17_todo/pullRequest?prid=6081322)